### DISCUSSION TITLE
0.8.0-rc1 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#608] [Announcements]
This is the first **release candidate** for version 0.8.0, and assuming no major bugs/regressions are found is likely to become 0.8.0-stable.

This release supports [**Godot 4.1**](https://github.com/godotengine/godot/releases/tag/4.1.1-stable).

## Changelog

These are the notable changes that have been made since [0.7.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.7.0-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.7.0-stable...v0.8.0-rc1).

Any breaking changes are denoted with ⚠️.

### Changed

- ⚠️ Changed collision layers and masks for `Area3D` to behave like they do in Godot Physics, allowing for asymmetrical setups, where overlaps are only reported if the mask of an `Area3D` contains the layer of the overlapping object.
- ⚠️ Changed the `body_set_force_integration_callback` method of `PhysicsServer3D` to behave like it does with Godot Physics, where omitting the binding of `userdata` requires that the callback also doesn't take any `userdata`. It also will no longer be called when the body is sleeping.

### Added

- Added timings of Jolt's various jobs to the "Physics 3D" profiler category.
- Added registering of `JoltPhysicsServer3D` as an actual singleton, which makes Jolt-specific server methods (e.g. `pin_joint_get_applied_force`) easier to deal with from dynamic scripting languages like GDScript.

### Fixed

- Fixed issue with `move_and_slide`, where under certain conditions you could get stuck on internal edges of a `ConcavePolygonShape3D` if the floor was within 5-ish degrees of `floor_max_angle`.
- Fixed issue with `move_and_slide`, where under certain conditions, while using a `BoxShape3D` or `CylinderShape3D` shape, you could get stuck on internal edges of a `ConcavePolygonShape3D`.
- Fixed issue where collision with `ConvexPolygonShape3D` could yield a flipped contact normal.
- Fixed issue where an `Area3D` with `monitoring` disabled wouldn't emit any entered events for already overlapping bodies once `monitoring` was enabled.
- Fixed issue where changing the center-of-mass of a `RigidBody3D` attached to a joint would shift its transform relative to the joint.
- Fixed issue where the `total_gravity` property on `PhysicsDirectBodyState3D` would always return a zero vector for kinematic bodies.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.8.0-rc1'>0.8.0-rc1</a>.</em>

**Comment by jams3223** [comment]
Let's goooo!

**Comment by Ughuuu** [comment]
Damn, amazing work as always, @mihe
